improver/categorical/utilities.py

```diff
@@ -241,19 +248,25 @@ def interrogate_decision_tree(decision_tree: Dict[str, Dict[str, Any]]) -> str:
     """
     # Diagnostic names and threshold values.
     requirements = {}
+    output = []
     for key, query in decision_tree.items():
         if not is_decision_node(key, query):
             continue
         diagnostics = get_parameter_names(
             expand_nested_lists(query, "diagnostic_fields")
         )
-        thresholds = expand_nested_lists(query, "diagnostic_thresholds")
-        for diagnostic, threshold in zip(diagnostics, thresholds):
-            requirements.setdefault(diagnostic, set()).add(threshold)
+        if "deterministic" in query:
+            for diagnostic in diagnostics:
+                output.append(f"\u26C5 {diagnostic} (deterministic)")
+                output.sort()
+        else:
+            thresholds = expand_nested_lists(query, "diagnostic_thresholds")
+            for diagnostic, threshold in zip(diagnostics, thresholds):
+                requirements.setdefault(diagnostic, set()).add(threshold)
 
     # Create a list of formatted strings that will be printed as part of the
```

Comment:
This comment needs to move with the `output = []` line.

Reply:
Done